neural-style-util.js

```diff
@@ -123,35 +124,17 @@ exports.saveImage = function(id, purpose, data, callback) {
 }
 
 exports.findImagePath = function(id, purpose, callback) {
-  var pathPrefix = exports.getImagePathPrefix(id, purpose);
-  function makeExtensionCheck(ext) {
-    return function(cb) {
-      var path = pathPrefix + '.' + ext;
-      fs.stat(path, function(err, stats) {
-        if (!err) {
-          cb(null, path);
-        } else {
-          cb(null, null);
-        }
-      });
-    };
-  }
-  async.parallel([
-    makeExtensionCheck('jpg'),
-    makeExtensionCheck('png'),
-  ], function(err, results) {
-    if (err) {
-      callback(err);
-      return;
-    }
-    var result = _.find(results, function(result) {
-      return result != null;
+  var fileNamePrefix = id + '_' + purpose;
+  var dataPath = config.get('dataPath');
+  fs.readdir(dataPath, function(err, results) {
+    var matches = results.filter(function(fileName) {
+      if (fileName.indexOf(fileNamePrefix) > -1) return true;
```

Comment:
I think we should really be checking that fileName starts with fileNamePrefix: fileName.indexOf(fileNamePrefix) == 0